### PR TITLE
Better choose when to build and upload Sentry info

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes: #4100

See also [discussion](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M4100.20Better.20choose.20when.20to.20build.20and.20upload.20Sentry.20info/near/1033355).

### CODE REVIEW

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -36,7 +36,6 @@
 		8BEA1EEE24A6983D00674040 /* ZLPConstants.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; name = ZLPConstants.m; path = ZulipMobile/ZLPConstants.m; sourceTree = "<group>"; };
 		8BEA1EEF24A6983D00674040 /* ZLPConstants.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = ZLPConstants.h; path = ZulipMobile/ZLPConstants.h; sourceTree = "<group>"; };
 		A148FEFB1E9D8CB900479280 /* zulip.mp3 */ = {isa = PBXFileReference; lastKnownFileType = audio.mp3; path = zulip.mp3; sourceTree = "<group>"; };
-		CF6CFE2C1E7DC55100F687C7 /* Build-Phases */ = {isa = PBXFileReference; lastKnownFileType = folder; path = "Build-Phases"; sourceTree = "<group>"; };
```
Comment (gnprice) [inline ios/ZulipMobile.xcodeproj/project.pbxproj:39]: This change LGTM but I think you have a stray line in the commit message:

    Build-Phases */ = {isa = PBXFileReferenc

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -271,7 +271,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if [[ $SDKROOT == *\"iPhoneSimulator\"* ]]; then\necho [Upload Debug Symbols to Sentry] Skipping sentry\nelse\nexport SENTRY_PROPERTIES=sentry.properties\n../node_modules/@sentry/cli/bin/sentry-cli upload-dsym\nfi\n";
```
Comment (gnprice) [inline ios/ZulipMobile.xcodeproj/project.pbxproj:276]: In addition to this, there's one other spot where we have a similar conditional: in the "Bundle React Native code and images" step.

**Review by gnprice — COMMENTED** [review]
Thanks -- this solution looks like a good one to me! Comments below; please merge at will after those.

**Comment by chrisbobbe** [comment]
> Thanks -- this solution looks like a good one to me! Comments below; please merge at will after those.

Merged, thanks for the review!

**Pull request merged**